### PR TITLE
feat: migrate time

### PR DESCRIPTION
**Pull request by Shinigami92**
This branch is based on #72

test with `npm run build && npx mocha test/time.unit.js`

### CODE REVIEW

File: src/time.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export class Time {
+  /**
+   * recent
+   *
+   * @method faker.time.recent
+   * @param outputType 'abbr' || 'wide' || 'unix' (default choice)
+   */
+  recent(outputType: 'abbr' | 'wide' | 'unix' = 'unix'): string | number {
+    let date: string | number | Date = new Date();
```
Comment (HonzaMac) [inline src/time.ts:9]: This ignores faker seed value. It would became non-deterministic.

**Review by HonzaMac — CHANGES_REQUESTED** [review]

File: src/time.ts (hunk 1 above)
Comment (Shinigami92) [reply src/time.ts:9]: I could add a TODO for later
We currently aim just for the migration and not for runtime changes

**Comment by MateusDadalto** [comment]
I don't think the actions are validating the typescript changes here.
If you run `tsc src/index.ts --outDir lib` and then `npm run test` we get tests failing.

File: src/time.ts (hunk 1 above)
Comment (HonzaMac) [reply src/time.ts:9]: Why adding feature then?

Comment (Shinigami92) [reply src/time.ts:9]: What do you mean?
This is just a rewrite of `lib/time.js`

Comment (prisis) [reply src/time.ts:11]: There is no new feaure  
  
\`\`\`

self.recent = _function_ (outputType) {  
  _if_ (_typeof_ outputType === 'undefined') { // this moved to the function  
    outputType = 'unix';  
  }  
  
  _var_ date = _new_ Date();  
  _switch_ (outputType) {  
    _case_ 'abbr':  
      date = date.toLocaleTimeString();  
      _break_;  
    _case_ 'wide':  
      date = date.toTimeString();  
      _break_;  
    _case_ 'unix':  
      date = date.getTime();  
      _break_;  
  }  
  _return_ date;  
};  
\`\`\`

Comment (Shinigami92) [reply src/time.ts:11]: Still don't understand, as you said there is no new feature right now in this PR
It's just a migration to TS and only touched compile time code

**Review by prisis — APPROVED** [review]

**Pull request merged**